pkg/providers/tinkerbell/hardware/etcd.go

```diff
@@ -36,17 +36,13 @@ func NewETCDReader(client client.Client) *ETCDReader {
 	}
 }
 
-// NewCatalogueFromETCD fetches the unprovisioned tinkerbell hardware objects and inserts in to ETCDReader catalogue.
-func (er *ETCDReader) NewCatalogueFromETCD(ctx context.Context) error {
+// HardwareFromETCD fetches the unprovisioned tinkerbell hardware objects and inserts in to ETCDReader catalogue.
+func (er *ETCDReader) HardwareFromETCD(ctx context.Context) error {
 	hwList, err := er.getUnprovisionedTinkerbellHardware(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to build catalogue: %v", err)
 	}
 
-	if len(hwList) == 0 {
-		return errors.New("no available hardware")
-	}
```

Comment:
Why is this not needed?

Reply:
I tested without it, so we run assertion `MinimumHardwareAvailableAssertionForCreate`. This also covers the case where there are no hardware available, so the explicit check became redundant.